File: testdata/fixtures/bench/perl/vulnerable/a03_sql_injection.pl
# Source: CWE-89 - SQL Injection via string interpolation in Perl
# Expected: BATOU-INJ, TAINT
# OWASP: A03:2021 - Injection (SQL Injection)

use strict;
use warnings;
use DBI;

my $username = $ARGV[0];
my $dbh = DBI->connect("dbi:mysql:database=app", "root", "");
my $query = "SELECT * FROM users WHERE name = '$username'";
my $sth = $dbh->prepare($query);
$sth->execute();

File: testdata/fixtures/bench/csharp/vulnerable/a03_sql_injection.cs
// Source: CWE-89 - SQL Injection via string concatenation in C#
// Expected: BATOU-CS
// OWASP: A03:2021 - Injection (SQL Injection)

using System;
using System.Data.SqlClient;

namespace App
{
    class UserRepository
    {
        public void FindUser(string username)
        {
            var conn = new SqlConnection("Server=localhost;Database=app;");
            var query = "SELECT * FROM users WHERE name = '" + username + "'";
            var cmd = new SqlCommand(query, conn);
            conn.Open();
            var reader = cmd.ExecuteReader();
        }

        static void Main(string[] args)
        {
            var repo = new UserRepository();
            repo.FindUser(args[0]);
        }
    }
}
